models/patientDelivery.js

```diff
@@ -108,20 +109,30 @@ module.exports = (sequelize, DataTypes) => {
           // fallthrough...
           break;
       }
-      const patientDeliveryUpdate = await sequelize.models.PatientDeliveryUpdate.create(
-        {
+      const [patientDeliveryUpdate, isCreated] = await sequelize.models.PatientDeliveryUpdate.findOrCreate({
+        where: {
           PatientDeliveryId: this.id,
           deliveryStatus,
+        },
+        defaults: {
           deliveryStatusDateTimeLocal: dateTimeLocal,
           CreatedById: userId,
           UpdatedById: userId,
         },
-        options
-      );
-      this.currentDeliveryStatus = deliveryStatus;
-      this.currentDeliveryStatusDateTimeLocal = dateTimeLocal;
-      this.UpdatedById = userId;
-      await this.save(options);
+        transaction: options?.transaction,
+      });
+      let updateCurrentDeliveryStatus = isCreated;
+      if (deliveryStatus === DeliveryStatus.RINGDOWN_RECEIVED && this.currentDeliveryStatus !== DeliveryStatus.RINGDOWN_SENT) {
```

Comment:
@billhoke actually, these conditionals check for and prevent overwriting later states with the received/confirmed states...